server/src/main/java/org/elasticsearch/cluster/routing/allocation/Explanations.java

```diff
@@ -0,0 +1,106 @@
+            within [%s] then Elasticsearch will not be able to allocate this shard to another node in the cluster.""";
+
+        public static final String EXISTING_STORES_FORBIDDEN = """
+            Elasticsearch is not permitted to allocate this shard to any of the nodes in the cluster that hold an in-sync copy of its \
+            data. If you expect this shard to be allocated to one of the nodes in the cluster then you should find this node in the \
+            node-by-node explanation and address the reasons which prevent Elasticsearch from allocating this shard there.""";
+
+        public static final String ALL_NODES_FORBIDDEN = """
+            Elasticsearch is not permitted to allocate this shard to any of the nodes in the cluster. If you expect this shard to be \
+            allocated to one of the nodes in the cluster then you should find this node in the node-by-node explanation and address the \
+            reasons which prevent Elasticsearch from allocating this shard there.""";
+
+    }
+
+    public static final class Rebalance {
+
+        public static final String YES = """
+            Elasticsearch can rebalance this shard onto a different node.""";
+
+        public static final String ALREADY_BALANCED = """
+            Elasticsearch cannot rebalance this shard onto a different node since there is no node to which allocation is permitted which \
+            would improve the cluster balance. If you expect this shard to be rebalanced onto one of the nodes in the cluster then you \
+            should find this node in the node-by-node explanation and address the reasons which prevent Elasticsearch from rebalancing \
+            this shard there.""";
+
+        public static final String AWAITING_INFO = """
+            Elasticsearch is currently retrieving information about the shard from one or more nodes, and will make a rebalancing decision \
+            once it receives the missing information. Please wait.""";
+
+        public static final String CLUSTER_THROTTLE = """
+            Elasticsearch is busy with other activities, but expects to be able to rebalance this shard when those activities finish. \
+            Please wait.""";
+
+        public static final String NODE_THROTTLE = """
+            Elasticsearch expects to be able to rebalance this shard onto another node, but the target node is currently busy with other \
+            activities. Please wait.""";
+
+        public static final String CANNOT_REBALANCE_CAN_ALLOCATE = """
+            Elasticsearch is not permitted to rebalance this shard onto any of the nodes in the cluster, even though it is permitted to \
+            allocate this shard onto one of those nodes. If you expect this shard to be rebalanced onto one of the nodes in the cluster \
+            then you should find this node in the node-by-node explanation and address the reasons which prevent Elasticsearch from \
+            rebalancing this shard there.""";
```

Comment:
```suggestion
            Although Elasticsearch is allowed to allocate this shard to another node, it isn't allowed to rebalance the shard. If you /
            expect the shard to be rebalanced another node, review the node-by-node explanation and address the reasons which prevent /
            Elasticsearch from rebalancing the shard.""";
```

Reply:
I was also wondering about leading with the "allowed to allocate" bit and whether this would be confusing because the important fact is that it's not allowed to rebalance there. But since you suggested it too, I've reordered things as you suggest.